src/combinatorics/rado/rank_function_independent_global.lean

```diff
@@ -0,0 +1,182 @@
+/-
+Copyright (c) 2023 Michael Stoll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Stoll
+-/
+
+import combinatorics.rado.rank_function_independent
+import category_theory.cofiltered_system
+
+/-!
+# Extension from locally independent to globally independent sections
+
+We use a compactness argument (essentially, that a product of discrete finite spaces
+is compact) to show that an independent section exists if there are independent sections
+on all finite subsets. See `rank_fn.independent.section`.
+
+We also prove that an independent section `f` on `s` can be extended to `insert i s`
+when the union of `f '' s` and `F i` has rank larger than `#s`; see
+`rank_fn.independent_on.extends`.
+-/
+
+namespace rank_fn
+
+open finset opposite
+
+universes u v
+
+variables {ι : Type u} {α : Type v} [decidable_eq α]
+
+-- We take inspiration from `combinatorics.hall.basic`.
+
+/-- The type of independent sections on a finite subset `s` of `ι` -/
+def independent_sections_on (r : rank_fn α) (F : ι → finset α) (s : finset ι) : Type (max u v) :=
+{f : s → α // independent r f ∧ ∀ ⦃i⦄ (hi : i ∈ s), f ⟨i, hi⟩ ∈ F i}
```

Comment:
I would imagine this is easier to use as
```suggestion
{f : s → α // independent r f ∧ ∀ i : s, f i ∈ F i}
```

Reply:
I don't think it really  makes a difference. Compare [this commit](https://github.com/leanprover-community/mathlib/pull/18789/commits/3b77e3a6375b13fc23ba2ce8945dd3f99d3a7d32) where I replaced `is_section ...` (which unfolded to what you suggest) with the current version. (And anyway, everything to do with `rado_functor` is not really meant for use outside the proof of `rank_fn.independent.section`.)